### ISSUE TITLE
On complete method

### ISSUE THREAD

**Issue by mikebski** [lian-yue/vue-upload-component#377]
How do you do something when all files are uploaded?

Watch does not really work on $refs so I can't watch for `uploaded` to be true.

Is there a solution?

**Comment by lian-yue** [comment]
```js
mounted: function () {
  $this.nextTick(function() {
    if (this.$refs.upload ) {
      this.$refs.upload.$watch('uploaded', function(newValue, oldValue) {
    })
    }
  })
}
```

Can you try

**Comment by mikebski** [comment]
Trying this - (`this.$nextTick` instead of what you have there) but it's not working:

```
mounted: function () {
    this.$nextTick(function() {
      if (this.$refs.upload ) {
        this.$refs.upload.$watch('uploaded', function(newValue, oldValue) {
          alert(`- Upload changed ${newValue} old was ${oldValue}`)
        })
      }
    })
  },

```

**Comment by mikebski** [comment]
OK, so looking at this, all my problems were because I was trying to show this on a page with a modal using bootstra-vue

Turns out the way I was using it was causing the ref to not get set when mount was called, so it was not working.

I also had drag/drop issues using this component in the modal.

My solution was to add this as it's own component in its own page and just redirect via the router when the upload is done, rather than show/hide a modal.

Closing because I figured out the issue, the watch code does actually work as specified when not putting this thing on a vue-bootstrap modal.
